internal/pkg/manifest/validate.go

```diff
@@ -1721,6 +1754,79 @@ func validateDepsForEssentialContainers(deps map[string]containerDependency) err
 	return nil
 }
 
+func validateExposedPorts(opts validateExposedPortsOpts) error {
+	exposedPorts := make(map[uint16]string)
+
+	if opts.mainContainerPort != nil {
+		exposedPorts[aws.Uint16Value(opts.mainContainerPort)] = opts.mainContainerName
+	}
+
+	for name, sidecar := range opts.sidecarConfig {
+		if sidecar.Port == nil {
+			continue
+		}
+		sidecarPort, protocol, err := ParsePortMapping(sidecar.Port)
+		if err != nil {
+			return err
+		}
+		if protocol != nil {
+			protocolVal := aws.StringValue(protocol)
+			var isValidProtocol bool
+			for _, valid := range nlbValidProtocols {
+				if strings.EqualFold(protocolVal, valid) {
+					isValidProtocol = true
+					break
+				}
+			}
+			if !isValidProtocol {
+				return fmt.Errorf(`invalid protocol %s; valid protocols include %s`, protocolVal, english.WordSeries(nlbValidProtocols, "and"))
+			}
+		}
+
+		port, err := strconv.ParseUint(aws.StringValue(sidecarPort), 10, 16)
+		if err != nil {
+			return err
+		}
+		if _, ok := exposedPorts[uint16(port)]; ok {
+			return &errContainersExposingSamePort{
+				firstContainer:  name,
+				secondContainer: exposedPorts[uint16(port)],
+				port:            uint16(port),
+			}
+		}
+		exposedPorts[uint16(port)] = name
+	}
+
+	// This condition takes care of the use case where target_container is set to x container and
+	// target_port exposing port 80 which is already exposed by container y.That means container x
+	// is trying to expose the port that is already being exposed by container y, so error out.
+	if opts.alb != nil {
+		alb := opts.alb
+		if alb.TargetPort == nil {
+			return nil
+		}
+		containerName := opts.mainContainerName
+		existingContainerName := exposedPorts[aws.Uint16Value(alb.TargetPort)]
+
+		if existingContainerName != "" {
+			containerName = existingContainerName
+			if alb.TargetContainer != nil {
+				if containerName != aws.StringValue(alb.TargetContainer) {
+					return &errContainersExposingSamePort{
+						firstContainer:  aws.StringValue(alb.TargetContainer),
+						secondContainer: containerName,
+						port:            aws.Uint16Value(alb.TargetPort),
+					}
+				}
+			}
+		} else if alb.TargetContainer != nil {
+			containerName = aws.StringValue(alb.TargetContainer)
+		}
+		exposedPorts[aws.Uint16Value(alb.TargetPort)] = containerName
+	}
+	return nil
```

Comment:
how about NLB validation?

Reply:
I am cutting a new PR for adding NLB validation and adding NLB exposed ports to the []ExposedPort slice. I had ask the same question earlier if we should add NLB config to the same PR or as a separate PR.

Reply:
🤦 sorry my bad, I forgot we discussed that earlier, sounds good!